multi-node-testkit/src/main/scala/org/apache/pekko/remote/testconductor/RemoteConnection.scala

```diff
@@ -94,44 +95,88 @@ private[pekko] case object Client extends Role
+  def apply(
+      role: Role,
+      sockaddr: InetSocketAddress,
+      poolSize: Int,
+      handler: ChannelInboundHandler): RemoteConnection = {
     role match {
       case Client =>
-        val socketfactory =
-          new NioClientSocketChannelFactory(Executors.newCachedThreadPool, Executors.newCachedThreadPool, poolSize)
-        val bootstrap = new ClientBootstrap(socketfactory)
-        bootstrap.setPipelineFactory(new TestConductorPipelineFactory(handler))
-        bootstrap.setOption("tcpNoDelay", true)
-        bootstrap.connect(sockaddr).getChannel
+        val bootstrap = new Bootstrap()
+        val eventLoopGroup = new NioEventLoopGroup(poolSize)
+        val cf = bootstrap
+          .group(eventLoopGroup)
+          .channel(classOf[NioSocketChannel])
+          .handler(new TestConductorPipelineFactory(handler))
+          .option[java.lang.Boolean](ChannelOption.TCP_NODELAY, true)
+          .option[java.lang.Boolean](ChannelOption.SO_KEEPALIVE, true)
+          .connect(sockaddr)
+        new RemoteConnection {
+          override def channelFuture: ChannelFuture = cf
+
+          override def shutdown(): Unit = {
+            try {
+              channelFuture.channel().close().sync()
+              eventLoopGroup.shutdownGracefully(1L, 1L, TimeUnit.SECONDS)
+            } catch {
+              case NonFatal(_) => // silence this one to not make tests look like they failed, it's not really critical
+            }
+          }
+        }
+
       case Server =>
-        val socketfactory =
-          new NioServerSocketChannelFactory(Executors.newCachedThreadPool, Executors.newCachedThreadPool, poolSize)
-        val bootstrap = new ServerBootstrap(socketfactory)
-        bootstrap.setPipelineFactory(new TestConductorPipelineFactory(handler))
-        bootstrap.setOption("reuseAddress", !Helpers.isWindows)
-        bootstrap.setOption("child.tcpNoDelay", true)
-        bootstrap.bind(sockaddr)
+        val bootstrap = new ServerBootstrap()
+        val parentEventLoopGroup = new NioEventLoopGroup(poolSize)
+        val childEventLoopGroup = new NioEventLoopGroup(poolSize)
+        val cf = bootstrap
+          .group(parentEventLoopGroup, childEventLoopGroup)
+          .channel(classOf[NioServerSocketChannel])
+          .childHandler(new TestConductorPipelineFactory(handler))
+          .option[java.lang.Boolean](ChannelOption.SO_REUSEADDR, !Helpers.isWindows)
+          .option[java.lang.Integer](ChannelOption.SO_BACKLOG, 2048)
+          .childOption[java.lang.Boolean](ChannelOption.TCP_NODELAY, true)
+          .childOption[java.lang.Boolean](ChannelOption.SO_KEEPALIVE, true)
+          .bind(sockaddr)
+          .sync()
+
+        new RemoteConnection {
+          override def channelFuture: ChannelFuture = cf
+
+          override def shutdown(): Unit = {
+            try {
+              channelFuture.channel().close().sync()
+              parentEventLoopGroup.shutdownGracefully(1L, 1L, TimeUnit.SECONDS)
```

Comment:
Same as previous comment about configuration

Reply:
I agree with @mdedetrich - this would be better as a setting in reference.conf

Reply:
I just updated the code without this, but which may waiting max to 15 seconds. the old code is deprecated.

Reply:
@mdedetrich I updated it to `.shutdownGracefully(0L, 0L, TimeUnit.SECONDS)` which alian the old behavior.